pkg/sql/logictest/testdata/logic_test/alter_column_type

```diff
@@ -161,3 +161,197 @@ ALTER TABLE t ALTER s TYPE STRING
+2  2  2
+3  3  3
+4  4  4
+
+statement ok
+CREATE TABLE t4 (x INT, y INT, z INT)
+
+statement ok
+INSERT INTO t4 VALUES (1, 2, 3)
+
+statement ok
+ALTER TABLE t4 ALTER COLUMN y TYPE STRING
+
+query TTBTTTB
+SHOW COLUMNS FROM t4
+----
+x      INT8    true   NULL            ·  {}         false
+y      STRING  true   NULL            ·  {}         false
+z      INT8    true   NULL            ·  {}         false
+rowid  INT8    false  unique_rowid()  ·  {primary}  true
+
+statement ok
+BEGIN
+
+statement ok
+CREATE TABLE t5 (x INT, y INT)
+
+statement error pq: unimplemented: performing an ALTER TABLE ... ALTER COLUMN TYPE in the same txn as the table was created is not supported
+ALTER TABLE t5 ALTER COLUMN x TYPE STRING
+
+statement ok
+ROLLBACK
+
+statement ok
+CREATE TABLE t6 (x TIMESTAMPTZ(6));
+INSERT INTO t6 VALUES ('2016-01-25 10:10:10.555555-05:00');
+INSERT INTO t6 VALUES ('2016-01-26 10:10:10.555555-05:00');
+ALTER TABLE t6 ALTER COLUMN x TYPE TIMESTAMPTZ(3);
+INSERT INTO t6 VALUES ('2016-01-26 10:10:10.55-05:00');
+
+query T
+SELECT * FROM t6 ORDER BY x
+----
+2016-01-25 16:10:10.556 +0100 CET
+2016-01-26 16:10:10.55 +0100 CET
+2016-01-26 16:10:10.556 +0100 CET
+
+# ensure column families stay the same
+statement ok
+CREATE TABLE t7(id INT, id2 INT, FAMILY f1 (id), FAMILY f2 (id2));
+INSERT INTO t7 VALUES (1), (2), (3);
+ALTER TABLE t7 ALTER COLUMN id2 TYPE STRING;
+
+query TT
+SHOW CREATE TABLE t7
+----
+t7      CREATE TABLE t7 (
+              id INT8 NULL,
+              id2 STRING NULL,
+              FAMILY f1 (id, rowid),
+              FAMILY f2 (id2)
+            )
+
+# ensure default expressions are casted correctly.
+statement ok
+CREATE TABLE t8 (x INT DEFAULT 1, y INT);
+INSERT INTO t8 (y) VALUES (1), (2), (3);
+ALTER TABLE t8 ALTER COLUMN x TYPE DATE;
+INSERT INTO t8 (y) VALUES (4);
+
+query TI
+SELECT * FROM t8 ORDER BY y
+----
+1970-01-02 00:00:00 +0000 +0000  1
+1970-01-02 00:00:00 +0000 +0000  2
+1970-01-02 00:00:00 +0000 +0000  3
+1970-01-02 00:00:00 +0000 +0000  4
+
+# ensure a runtime error rollsback correctly.
+statement ok
+
+statement ok
+CREATE TABLE t9 (x INT PRIMARY KEY)
+
+statement error pq: unimplemented: alter column type requiring rewrite of on-disk data is currently not supported for columns that are part of an index
+ALTER TABLE t9 ALTER COLUMN x TYPE STRING
+
+statement ok
+CREATE TABLE t10 (x INT, y INT, INDEX(x, y))
+
+statement error pq: unimplemented: alter column type requiring rewrite of on-disk data is currently not supported for columns that are part of an index
+ALTER TABLE t10 ALTER COLUMN y TYPE STRING
+
+statement ok
+CREATE TABLE t11 (x INT)
+
+statement error pq: unimplemented: alter column type using expression is not supported\nHINT: You have attempted to use a feature that is not yet implemented.\nSee: https://github.com/cockroachdb/cockroach/issues/47706
+ALTER TABLE t11 ALTER COLUMN x TYPE STRING USING CAST(x AS STRING)
+
+# dropping a column that owns a sequence should not be allowed.
```

Comment:
On this note, can you see if we need to disable column type changes on columns that are used in views?

Reply:
Tested views, they should be okay, they use the table id and not column id and the selects work as expected with the correct type.